recipe/meta.yaml

```diff
@@ -1,46 +1,90 @@
 {% set name = "splink" %}
-{% set version = "3.6.0" %}
+{% set version = "3.9.10" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 6bbef574b12ebf74e112960e2bc5d8baf3ca60f5d889a581dafb2f29173e90a8
+  sha256: 891996d3eaa6ff58cc3e3cd97c0aa0c45246a8b1c858cd4fe77f617e190f20ee
 
 build:
-  noarch: python
-  number: 1
-  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [py<38 or s390x]
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
+# the requirements refer to the duckdb-less installation documented at
+# https://moj-analytical-services.github.io/splink/getting_started.html#duckdb-less-installation
 requirements:
   host:
-    - python >=3.7,<4.0
+    - python
     - pip
-    - poetry >=0.12
+    - setuptools
+    - wheel
+    - poetry
+    - altair >=5.0.1
+    - attrs >=23.1.0
+    - greenlet >=2.0.2
+    - importlib_resources >=5.4.0
+    - jinja2 >=3.1.2
+    - jsonschema >=4.17.3
+    - markupsafe >=2.1.3
+    - pandas >=1.3.5
+    - phonetics >=1.0.5
+    - pkgutil-resolve-name >=1.3.10  #[py<=38]
+    - psycopg2 >=2.9.7
+    - py4j >=0.10.9.7
+    - pyrsistent >=0.19.3
+    - pyspark >=3.4.1
+    - python-dateutil >=2.8.2
+    - pytz >=2023.3.post1
+    - six >=1.16.0
+    - sqlalchemy >=1.4.49
+    - sqlglot >=11.4.1
+    - toolz >=0.12.0
+    - typing-extensions >=4.7.1
+    - zipp >=3.15.0
   run:
-    - python >=3.7,<4.0
+    - python
     - jsonschema >=3.2,<5.0
-    - python-duckdb >=0.6.0,<0.7.0
-    - pandas >=1.0.0,<2.0.0
-    - sqlglot >=5.1.0
-    - altair >=4.2.0
-    - jinja2 >=3.0.3,<4.0.0
+    - pandas >=1.3.0
+    - sqlglot >=13.0.0,<19.0.0
+    - altair >=5.0.1,<6.0.0
+    - jinja2 >=3.0.3
+    - phonetics >=1.0.5,<2.0.0
   run_constrained:
     - pyspark >=3.2.1
+    - sqlalchemy >=1.4.0,<2.0.0
+    - psycopg2 >=2.8.0
 
 test:
   imports:
     - splink
+  commands:
+  # while doing pip check got an error: splink 3.9.10 requires duckdb, which is not installed.
+  #  - pip check 
```

Comment:
It might be worth running the tests. But they are not included in the pypi archive.